R/pkg-knitr.R

```diff
@@ -0,0 +1,170 @@
+#' knit print polars DataFrame
+#'
+#' Mimics python-polars' NotebookFormatter
+#' for HTML outputs.
+#'
+#' Outputs HTML tables if the output format is HTML
+#' and the document's `df_print` option is not `"default"` or `"tibble"`.
+#'
+#' Or, the output format can be enforced with R's `options` function as follows:
+#'
+#' - `options(polars.df_print = "default")` for the default print method.
+#' - `options(polars.df_print = "html")` for the HTML table.
+#' @name knit_print.DataFrame
+#' @param x a polars DataFrame to knit_print
+#' @param ... additional arguments, not used
+#' @keywords DataFrame
+#' @export
+knit_print.DataFrame <- function(x, ...) {
+  .print_opt <- getOption("polars.df_print", "auto")
+  .rmd_df_print <- knitr::opts_knit$get("rmarkdown.df_print")
+  if (.print_opt == "html" ||
+    (.print_opt != "default" &&
+      !isTRUE(.rmd_df_print %in% c("default", "tibble")) &&
+      knitr::is_html_output())) {
+    x |>
+      to_html_table() |>
+      knitr::raw_html() |>
+      knitr::asis_output()
+  } else {
+    print(x)
+  }
+}
+
+#' Generate HTML table from a DataFrame
+#' @param x DataFrame
+#' @param max_cols an integer of maximum number of columns to display
+#' @param max_rows an integer of maximum number of rows to display
+#' @return A string of HTML code
+#' @examples
+#' to_html_table(mtcars, 3, 3)
+#' @noRd
+#' @importFrom utils getFromNamespace
+to_html_table <- function(x, max_cols = 75, max_rows = 40) {
+  escape_html <- getFromNamespace("escape_html", "knitr")
```

Comment:
Are you 100% sure that no user will ever want to execute this function without `knitr` installed? If that's not the case, we should have a conditional as described here: https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Suggested-packages

Reply:
Makes sense. I add lines to check which knitr is installed.